src/main/apiClient.ts

```diff
@@ -694,15 +694,24 @@ export default class ApiClient {
     return this.http.post('/unlink', { body: data, accessToken })
   }
 
-  refreshTokens({ accessToken }: { accessToken: string }): Promise<AuthResult> {
+  refreshTokens({ accessToken, refreshToken }: { accessToken?: string, refreshToken?: string }): Promise<AuthResult> {
```

Comment:
I know reusing the same method is what's been decided in the tech analysis, but perhaps having two separate methods with one being deprecated would have been cleaner? Before this change, the `accessToken` parameter was mandatory as the function signature indicated. Now both `accessToken` and `refreshToken` can be undefined when calling the function, which is plain wrong.

Reply:
I also thought the same but we finally decided to go for one method instead of two. 
If the method is called with no token, the endpoint /oauth/token will respond with a 400 request allowing the client to understand that the problem was due to an empty form. But you are right, one method instead of two can lead to problems and is not super clean.

Reply:
This can be handled with a sum type for the parameters:
```typescript
// L149
export type RefreshTokenParams = { accessToken: string } | { refreshToken: string, scope?: Scopes }

refreshTokens(params: RefreshTokenParams): Promise<AuthResult> {
  const result =
    ('refreshToken' in params)
      ? this.http.post<AuthResult>(this.tokenUrl, {
          body: {
            clientId: this.config.clientId,
            grantType: 'refresh_token',
            refreshToken: params.refreshToken,
            ...pick(params, 'scope'),
          }
        })
      : this.http.post<AuthResult>('/token/access-token', {
          body: {
            clientId: this.config.clientId,
            accessToken: params.accessToken
          }
        })

    return result.then(enrichAuthResult)
  }
```
- don't forget the `scope` parameter ❕ 
- prefer matching on your type rather than performing null checks

Reply:
I tried something like this (with type matching and or operator) in order to protect against a request with no token. But the endpoint /token/access-token is called anyway (it responds with a 400 request). The type refreshTokenParams has the same behaviour.
Concerning the scope, I did not know where to put the scope. I noticed there is no scope parameter in documentation of oauth/token, I reported it to Alex. Thank you for feedback.